File: src/main/java/com/lbx/demo/model/TitlePrincipal.java
package com.lbx.demo.model;

import lombok.*;


/**
* 3. Get two actors and return all the titles in which both of them played at
**/
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class TitlePrincipal implements FieldSettable{
    private String titleId;
    private String nameId;
    private Integer ordering;
    private String category;
    private String job;
    private String characters;

    public boolean isActor() {
        return job.equals("actress") || job.equals("actor");
    }

    @Override
    public void setField(String name, String value) {
        switch (name) {
            case "tconst" -> titleId = value;
            case "nconst" -> nameId = value;
            case "ordering" -> ordering = Integer.valueOf(value);
            case "category" -> category = value;
            case "job" -> job = value;
            case "characters" -> characters = value;
        }
    }
}

File: src/main/rust/src/models/title_principal.rs
use serde::Serialize;

use crate::models::mapper::FieldSettable;


#[derive(Debug, Default, Clone, Hash, PartialEq, PartialOrd, Eq, Serialize)]
pub struct TitlePrincipal {
    pub title_id: String,
    pub name_id: String,
    pub ordering: i8,
    pub category: String,
    pub job: String,
    pub characters: String,
}

impl TitlePrincipal {
    pub fn is_actor(&self) -> bool {
        self.job.eq("actor") || self.job.eq("actress")
    }
}

impl FieldSettable for TitlePrincipal {
    fn new() -> Self {
        TitlePrincipal::default()
    }

    fn set_field(&mut self, name: &str, value: &str) {
        match name {
            "tconst" => self.title_id = value.to_string(),
            "nconst" => self.name_id = value.to_string(),
            "ordering" => self.ordering = value.parse().unwrap(),
            "category" => self.category = value.to_owned(),
            "job" => self.job = value.to_owned(),
            "characters" => self.characters = value.to_owned(),
            _ => {
                panic!("no such column![{name}]")
            }
        }
    }
}
